crates/typst/src/meta/mod.rs

```diff
@@ -0,0 +1,15 @@
+use crate::doc::{Lang, Region};
+use crate::model::StyleChain;
+
+/// The named with which an element is referenced.
+pub trait LocalName {
+    /// Get the name in the given language and (optionally) region.
+    fn local_name(lang: Lang, region: Option<Region>) -> &'static str
+    where
+        Self: Sized;
+
+    /// Resolve the local name with a style chain.
+    fn local_name_in(styles: StyleChain) -> &'static str
+    where
+        Self: Sized;
```

Comment:
The implementation of this is copy-pasted everywhere, which is unfortunate. I assume the reason is that TextElem isn't available here. However, there are multiple other options:
- Use `Self::elem().local_name(styles)`, although that returns an `Option`
- Create a blanket impl trait `LocalNameIn` in `typst-library`.

Reply:
That's a good idea, I'll change it to having a secondary trait in `typst-lib`